packages/compiler/core/parser.ts

```diff
@@ -1255,12 +1255,19 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
   function parseProjection(): ProjectionNode {
     const pos = tokenPos();
     const directionId = parseIdentifier("projectionDirection");
-    let direction: "from" | "to";
+    let direction: IdentifierNode<"to" | "from">;
     if (directionId.sv !== "from" && directionId.sv !== "to") {
       error({ code: "token-expected", messageId: "projectionDirection" });
-      direction = "from";
+      direction = {
+        kind: SyntaxKind.Identifier,
+        sv: "to",
+        pos: pos,
+        end: pos,
+        symbol: undefined as any,
+        flags: NodeFlags.Synthetic,
+      };
```

Comment:
Hmm, this part is a little complicated. I think it's probably simpler if we add `directionId: IdentifierNode` alongside `direction: "to" | "from"`.

Reply:
so  could also make direction optional because I think we'll want that in the future anyway. This would just move the defaulting to the checker.